### PR TITLE
Create pod to force volume provisioning in dynamic provisioning storage e2e test

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

The test creates a PVC and wait for it to be bound, however, it currently uses the cluster default StorageClass, which might be configured with `WaitForFirstConsumer` instead.

This issue was discovered in this job added recently: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-storage-selinux/1268090733013766146

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

CC @jsafrane @msau42

### CODE REVIEW

File: test/e2e/storage/volume_provisioning.go [hunk 1]
```diff
@@ -662,10 +662,12 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
 				ClaimSize:    "2Gi",
 				ExpectedSize: "2Gi",
 			}
-
+			suffix := fmt.Sprintf("volume-lifecycle")
+			test.Class = newStorageClass(test, ns, suffix)
```
Comment (msau42) [inline test/e2e/storage/volume_provisioning.go:666]: What if we change the test to handle delayed binding instead?

Comment (bertinatto) [reply test/e2e/storage/volume_provisioning.go:666]: Added a commit to handle delayed binding.

From what I understand, this test should create its own SC anyway, so I left the first patch as well.

Comment (msau42) [reply test/e2e/storage/volume_provisioning.go:666]: Then we're not actually testing the default StorageClass, which I think is the purpose of this test.

Comment (bertinatto) [reply test/e2e/storage/volume_provisioning.go]: I assumed the opposite, since other tests explicitly mention they use the default SC. Dropped this code.

**Comment by msau42** [comment]
/hold
Can you update the PR title?

**Comment by bertinatto** [comment]
/hold cancel
Updated title

**Pull request merged**